### PR TITLE
FastSim bug fix: make matched hits aware of order of components (76X, take 3)

### PR DESCRIPTION
**Pull request by lveldere**
replaces #10111 ...

### CODE REVIEW

**Comment by lveldere** [comment]
please test

File: DataFormats/TrackerRecHit2D/src/classes_def.xml [hunk 1]
```diff
@@ -11,12 +11,13 @@
...
-  <class name="SiTrackerGSMatchedRecHit2D" ClassVersion="14">
-    <version ClassVersion="14" checksum="3882739671"/>
-    <version ClassVersion="13" checksum="1553184391"/>
-    <version ClassVersion="12" checksum="2874379530"/>
-    <version ClassVersion="11" checksum="1553184391"/>
-    <version ClassVersion="10" checksum="3741285161"/>
+  <class name="SiTrackerGSMatchedRecHit2D" ClassVersion="15">
+   <version ClassVersion="15" checksum="3518158739"/>
+   <version ClassVersion="14" checksum="1823455267"/>
+   <version ClassVersion="13" checksum="1553184391"/>
```
Comment (slava77) [inline DataFormats/TrackerRecHit2D/src/classes_def.xml]: How did it happen that v11 and v13 have the same class version?
Do we ever read these objects from files?

Comment (lveldere) [inline DataFormats/TrackerRecHit2D/src/classes_def.xml]: argh
I'll study the class version this afternoon,
and make sure all is as it should be.
Perhaps this V12 was something intermediate...
Lukas

**Comment by lveldere** [comment]
+1

File: DataFormats/TrackerRecHit2D/src/classes_def.xml (hunk 1 above)
Comment (lveldere) [inline DataFormats/TrackerRecHit2D/src/classes_def.xml]: This is the pr where v13 was introduced
https://github.com/cms-sw/cmssw/pull/8654
The test compiled, so I guess it worked?
This V13 will never be used in production, so I would not bother about it.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/10119	e1e4519
- full reco is not affected by changes made in this PR
- changes  in the fastsim are similar to the observations made in https://github.com/cms-sw/cmssw/pull/10079#issuecomment-119749894

**Comment by davidlange6** [comment]
+1

**Pull request merged**